src/server/middleware/createRequestHtmlFragment.jsx

```diff
@@ -69,11 +77,22 @@ export default function createRequestHtmlFragment({ createRoutes }) {
           },
         }));
 
-      await dispatch(composeModules(routeModules));
-
       const state = store.getState();
       const disableScripts = state.getIn(['rendering', 'disableScripts']);
       const renderPartialOnly = state.getIn(['rendering', 'renderPartialOnly']);
+
+      if (disableScripts || renderPartialOnly) {
+        await dispatch(composeModules(routeModules));
+      } else {
+        const fallback = await breaker.fire({ dispatch, modules: routeModules });
```

Comment:
maybe rename breaker to show its related to getModuleData

```suggestion
        const fallback = await getModuleDataBreaker.fire({ dispatch, modules: routeModules });
```